### PR TITLE
fix: Sentry.framework does not support the minimum OS Version specified in the Info.plist

### LINKED ISSUE #3769: Sentry.framework does not support the minimum OS Version specified in the Info.plist.
### Platform

iOS

### Environment

TestFlight

### Installed

Swift Package Manager

### Version

8.22.2

### Did it work on previous versions?

yes but unsure which version. maybe 8.21.0

### Steps to Reproduce

1. Using XCode 15.3, create a new project and add Sentry via SPM. The minimum deployment should be iOS 17.0
2. Archive the project and validate it 

### Expected Result

Validation should succeed with no issues 

### Actual Result

The bundle {appName}.app/Frameworks/Sentry.framework does not support the minimum OS Version specified in the Info.plist.

### Are you willing to submit a PR?

_No response_

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Static framework was failing Xcode validation because of minimum os version.

## :bulb: Motivation and Context

close #3769

### CODE REVIEW

File: scripts/build-xcframework.sh [hunk 1]
```diff
@@ -19,8 +21,13 @@ generate_xcframework() {
     for sdk in "${sdks[@]}"; do
         if [[ -n "$(grep "${sdk}" <<< "$ALL_SDKS")" ]]; then
             xcodebuild archive -project Sentry.xcodeproj/ -scheme "$scheme" -configuration Release -sdk "$sdk" -archivePath ./Carthage/archive/${scheme}${sufix}/${sdk}.xcarchive CODE_SIGNING_REQUIRED=NO SKIP_INSTALL=NO CODE_SIGN_IDENTITY= CARTHAGE=YES MACH_O_TYPE=$MACH_O_TYPE
-            
+                 
             createxcframework+="-framework Carthage/archive/${scheme}${sufix}/${sdk}.xcarchive/Products/Library/Frameworks/${scheme}.framework "
+
+            if [ "$MACH_O_TYPE" = "staticlib" ]; then
+                local infoPlist="Carthage/archive/${scheme}${sufix}/${sdk}.xcarchive/Products/Library/Frameworks/${scheme}.framework/Info.plist"
+                plutil -replace "MinimumOSVersion" -string "9999" "$infoPlist"
```
Comment (philipphofmann) [inline scripts/build-xcframework.sh:29]: `m`: Please add a comment to explain this fix.

Comment (philipphofmann) [inline scripts/build-xcframework.sh:29]: `h`: Does that have other implications for Xcode versions that know how to parse the `MinimumOSVersion` correctly?

**Review by philipphofmann — COMMENTED** [review]
Thanks @brustolin. One question.

File: scripts/build-xcframework.sh (hunk 1 above)
Comment (brustolin) [reply scripts/build-xcframework.sh:33]: I dont know.

I have tried it for iOS, Catalyst, tvOS.
I validated with the app store for all those above. 
Everything is working with this change.

Comment (philipphofmann) [reply scripts/build-xcframework.sh:35]: Okay, if you validated it, we should be fine.

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**